### PR TITLE
dev-python/pyzbar: bump version to 0.1.8

### PR DESCRIPTION
**Pull request by linxon**
Closes: https://bugs.gentoo.org/696660
Package-Manager: Portage-2.3.84, Repoman-2.3.16
Signed-off-by: Yury Martynov <email@linxon.ru>

### CODE REVIEW

File: dev-python/pyzbar/pyzbar-0.1.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		virtual/python-unittest-mock[${PYTHON_USEDEP}]
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+python_test() {
+	"${EPYTHON}" -m unittest discover -v || die "Testing failed with ${EPYTHON}"
+}
```
Comment (juippis) [inline dev-python/pyzbar/pyzbar-0.1.8.ebuild]: Can this be replaced with `distutils_enable_tests`?

File: dev-python/pyzbar/pyzbar-0.1.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RDEPEND="
+	dev-python/pillow[${PYTHON_USEDEP}]
+	media-gfx/zbar
+	virtual/python-enum34[${PYTHON_USEDEP}]
+	virtual/python-pathlib[${PYTHON_USEDEP}]
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/pyzbar/pyzbar-0.1.8.ebuild]: Most likely wrong, *distutils-r1* already puts this in BDEPEND.

File: dev-python/pyzbar/pyzbar-0.1.8.ebuild (hunk 1 above)
Comment (linxon) [reply dev-python/pyzbar/pyzbar-0.1.8.ebuild]: okay

Comment (linxon) [reply dev-python/pyzbar/pyzbar-0.1.8.ebuild]:
```bash
>>> Source compiled.
Unable to configure loopback interface: Operation not permitted
>>> Test phase: dev-python/pyzbar-0.1.8
 * python3_6: running distutils-r1_run_phase python_test
test_bounding_box (pyzbar.tests.test_locations.TestLocations) ... ok
test_convex_duplicates (pyzbar.tests.test_locations.TestLocations) ... ok
test_convex_hull_empty (pyzbar.tests.test_locations.TestLocations) ... ok
test_convex_square (pyzbar.tests.test_locations.TestLocations) ... ok
test_other (pyzbar.tests.test_locations.TestLocations) ... ok
test_decode_code128 (pyzbar.tests.test_pyzbar.TestDecode)
Read both barcodes in `code128.png` ... ok
test_decode_numpy (pyzbar.tests.test_pyzbar.TestDecode)
Read image using Pillow and convert to numpy.ndarray ... ok
test_decode_opencv (pyzbar.tests.test_pyzbar.TestDecode)
Read image using OpenCV ... skipped 'OpenCV not installed'
test_decode_qrcode (pyzbar.tests.test_pyzbar.TestDecode)
Read barcode in `qrcode.png` ... ok
test_decode_qrcode_rotated (pyzbar.tests.test_pyzbar.TestDecode)
Read barcode in `qrcode_rotated.png` ... ok
test_decode_tuple (pyzbar.tests.test_pyzbar.TestDecode)
Read barcodes in pixels ... ok
test_empty (pyzbar.tests.test_pyzbar.TestDecode)
Do not show any output for an image that does not contain a barcode ... ok
test_external_dependencies (pyzbar.tests.test_pyzbar.TestDecode)
External dependencies ... ok
test_inconsistent_dimensions (pyzbar.tests.test_pyzbar.TestDecode) ... /var/tmp/portage/dev-python/pyzbar-0.1.8/work/pyzbar-0.1.8/pyzbar/tests/test_pyzbar.py:188: DeprecationWarning: Please use assertRaisesRegex instead.
  decode, data
ok
test_symbols (pyzbar.tests.test_pyzbar.TestDecode)
Read only qrcodes in `qrcode.png` ... ok
test_symbols_not_present (pyzbar.tests.test_pyzbar.TestDecode)
Read only code128 in `qrcode.png` ... ok
test_unsupported_bits_per_pixel (pyzbar.tests.test_pyzbar.TestDecode) ... ok
test_unsupported_bpp (pyzbar.tests.test_pyzbar.TestDecode) ... ok
test_zbar_image_create_fail (pyzbar.tests.test_pyzbar.TestDecode) ... ok
test_zbar_image_scanner_create_fail (pyzbar.tests.test_pyzbar.TestDecode) ... ok
test_zbar_scan_image_fail (pyzbar.tests.test_pyzbar.TestDecode) ... ok
test_read_code128 (pyzbar.tests.test_read_zbar.TestReadZbar)
Read CODE 128 barcodes ... ok
test_read_qrcode (pyzbar.tests.test_read_zbar.TestReadZbar)
Read QR code ... ok
test_found_non_windows (pyzbar.tests.test_zbar_library.TestLoad)
zbar loaded ok on non-Windows platform ... ok
test_found_second_attempt_windows (pyzbar.tests.test_zbar_library.TestLoad)
zbar found on the second attempt on Windows ... ok
test_found_windows (pyzbar.tests.test_zbar_library.TestLoad)
zbar found on Windows ... ok
test_not_found_non_windows (pyzbar.tests.test_zbar_library.TestLoad)
zbar not found on non-Windows platform ... ok
test_not_found_windows (pyzbar.tests.test_zbar_library.TestLoad)
zbar not found on Windows ... ok
test_32bit (pyzbar.tests.test_zbar_library.TestWindowsFnames) ... ok
test_64bit (pyzbar.tests.test_zbar_library.TestWindowsFnames) ... ok

----------------------------------------------------------------------
Ran 30 tests in 0.334s

OK (skipped=1)
 * python3_6: running distutils-r1_run_phase _distutils-r1_clean_egg_info
>>> Completed testing dev-python/pyzbar-0.1.8
```

**Comment by linxon** [comment]
maybe I need to close this PR ?

because this package requires only `sys-auth/authenticator-0.2.4` in the gentoo tree
```bash
 * dependency graph for sys-auth/authenticator-0.2.4
 `--  sys-auth/authenticator-0.2.4  [~amd64 keyword] 
   `--  dev-lang/python-3.6.9  (dev-lang/python) amd64 
   `--  dev-lang/python-exec-2.4.6-r1  (>=dev-lang/python-exec-2) amd64  [python_targets_python3_6]
   `--  dev-lang/python-3.7.5-r1  (dev-lang/python) amd64 
   `--  app-crypt/libsecret-0.18.8  (app-crypt/libsecret) amd64 
   `--  dev-python/pillow-6.2.1  (dev-python/pillow) amd64  [python_targets_python3_6(-)? python_targets_python3_7(-)? -python_single_target_pypy3(-) -python_single_target_python2_7(-) -python_single_target_python3_8(-) python_single_target_python3_6(+)? python_single_target_python3_7(+)?]
   `--  dev-python/pygobject-3.32.1  (dev-python/pygobject) amd64  [python_targets_python3_6(-)? python_targets_python3_7(-)? -python_single_target_pypy3(-) -python_single_target_python2_7(-) -python_single_target_python3_8(-) python_single_target_python3_6(+)? python_single_target_python3_7(+)?]
   `--  dev-python/pyotp-2.2.6-r1  (dev-python/pyotp) [~amd64 keyword]  [python_targets_python3_6(-)? python_targets_python3_7(-)? -python_single_target_pypy3(-) -python_single_target_python2_7(-) -python_single_target_python3_8(-) python_single_target_python3_6(+)? python_single_target_python3_7(+)?]
   `--  dev-python/python-gnupg-0.4.5  (dev-python/python-gnupg) amd64  [python_targets_python3_6(-)? python_targets_python3_7(-)? -python_single_target_pypy3(-) -python_single_target_python2_7(-) -python_single_target_python3_8(-) python_single_target_python3_6(+)? python_single_target_python3_7(+)?]
   `--  dev-python/pyzbar-0.1.8  (dev-python/pyzbar) ~amd64  [python_targets_python3_6(-)? python_targets_python3_7(-)? -python_single_target_pypy3(-) -python_single_target_python2_7(-) -python_single_target_python3_8(-) python_single_target_python3_6(+)? python_single_target_python3_7(+)?]
   `--  media-libs/gd-2.2.5-r2  (media-libs/gd) amd64 
   `--  x11-libs/gtk+-3.24.11  (x11-libs/gtk+) amd64 
   `--  x11-apps/xhost-1.0.8  (x11-apps/xhost) amd64 
   `--  dev-util/meson-0.52.1  (>=dev-util/meson-0.51.2) amd64 
   `--  dev-util/ninja-1.8.2  (>=dev-util/ninja-1.8.2) amd64 
   `--  x11-base/xorg-server-1.20.6  (x11-base/xorg-server) amd64  [xvfb]
[ sys-auth/authenticator-0.2.4 stats: packages (16), max depth (1) ]
```

**Comment by linxon** [comment]
I just tried to bump [a new version](https://gitlab.gnome.org/World/Authenticator/-/tags/3.32.2) of the `sys-auth/authenticator` (https://bugs.gentoo.org/696476) and this version use media-gfx/zbar[python] bindings instead current latest `dev-python/pyzbar`

**Comment by linxon** [comment]
also after installing `sys-auth/authenticator-0.2.4` I get 
```bash
55df1632222e /tmp $ DISPLAY=':4' authenticator 
bash: /usr/bin/authenticator: /usr/bin/python3.6^M: bad interpreter: No such file or directory
```

**Comment by linxon** [comment]
oh. my mistake

it's req `pyzbar` also 

More: https://gitlab.gnome.org/World/Authenticator/blob/d6c8d61d2568d92d3ad7d1ae5c7743f9919fb2b8/src/Authenticator/models/qr_reader.py#L23 

the PR is actual yet

**Pull request closed without merging**